jujugui/static/gui/src/app/utils/bundle-importer.js

```diff
@@ -422,7 +437,16 @@ YUI.add('bundle-importer', function(Y) {
           ghostService.set('displayName', displayName);
 
           ghostService.set('config', config);
-
+          // Convert the string space delimeted constraints into an object
+          // as required by the juju api.
+          var constraints = {};
```

Comment:
I think this is already done by the env (see env.prepareConstraints) so not sure if it's worth redoing it now.
In general, I think it should be safe to pass both string and object formats to deploy().